Source/WebCore/dom/ViewTransition.cpp

```diff
@@ -859,12 +860,26 @@ RenderViewTransitionCapture* ViewTransition::viewTransitionNewPseudoForCapturedE
     return nullptr;
 }
 
+// https://drafts.csswg.org/css-view-transitions/#page-visibility-change-steps
+void ViewTransition::visibilityStateChanged()
+{
+    if (!document())
+        return;
+
+    if (protectedDocument()->hidden() && protectedDocument()->activeViewTransition() == this) {
+        protectedDocument()->clearInboundViewTransitionParams();
+        skipViewTransition(Exception { ExceptionCode::InvalidStateError, "Skipping view transition because document visibility state has become hidden."_s });
+    } else
+        ASSERT(!protectedDocument()->activeViewTransition());
```

Comment:
Looks like this fails sometimes.

My guess is that this happens because it's possible for multiple ViewTransition objects to be alive at the same time, even though only one is active. We're then running this visibility state change algorithm multiple times at once.

```suggestion
    if (protectedDocument()->hidden()) {
        if (protectedDocument()->activeViewTransition() == this)
            skipViewTransition(Exception { ExceptionCode::InvalidStateError, "Skipping view transition because document visibility state has become hidden."_s });
    } else
        ASSERT(!protectedDocument()->activeViewTransition());
```